### PR TITLE
Add web-client compatibility, build process and structure for NodeJS

### PR DESCRIPTION
**Pull request by sisou**
## What's in this pull request?

Support for running the WASM light client in NodeJS, both as a CommonJS and ESM module.

This PR does not touch any Rust files, only re-organizes JS/TS and supporting files inside the `web-client` crate.

## Pull request checklist

- [x] All tests pass. The project builds and runs.
- [x] I have resolved any merge conflicts.
- [x] I have resolved all `clippy` and `rustfmt` warnings.

### CODE REVIEW

**Comment by Eligioo** [comment]
@sisou I think the example in the web-client doesn't work anymore after this restructure:

```
 Failed to load resource: the server responded with a status of 404 (File not found): :8000/dist/lib/brows…/client-proxy.mjs:1 
 Failed to load resource: the server responded with a status of 404 (File not found): :8000/dist/lib/brows…sfer-handlers.mjs:1
```

**Comment by sisou** [comment]
> @sisou I think the example in the web-client doesn't work anymore after this restructure:
> 
> ```
>  Failed to load resource: the server responded with a status of 404 (File not found): :8000/dist/lib/brows…/client-proxy.mjs:1 
>  Failed to load resource: the server responded with a status of 404 (File not found): :8000/dist/lib/brows…sfer-handlers.mjs:1
> ```

Did you build those scripts with `./scripts/build-launcher.sh`?

**Comment by Eligioo** [comment]
> Did you build those scripts with `./scripts/build-launcher.sh`?

Nope I was not aware that this now has become necessary. It works again now for the browser after running the `build-launcher`. Then the README should only then be updated. @sisou

**Comment by sisou** [comment]
I added a README in the `example` folder, and fixed the installation for the NodeJS example.

File: web-client/dist/README.md [hunk 1]
```diff
@@ -1,9 +1,9 @@
-# Nimiq Albatross Web Client
+# Nimiq Albatross Light Client
 
-A very light Nimiq Proof-of-Stake client that runs in web browsers, compiled from Rust to WebAssembly.
+A very light Nimiq Proof-of-Stake client browsers and NodeJS, compiled from Rust to WebAssembly.
```
Comment (jsdanielh) [inline web-client/dist/README.md]:
```suggestion
A very light Nimiq Proof-of-Stake client for browsers and NodeJS, compiled from Rust to WebAssembly.
```

**Review by jsdanielh — APPROVED** [review]

**Pull request merged**